api/src/main/resources/application_detailed_summary.py

```diff
@@ -0,0 +1,101 @@
+import time
+import json
+import logging
+import sys
+from importlib import import_module
+
+from summary_aggregator import ComponentSummaryAggregator
+from plugins_summary.yarn_connection import YarnConnection
+from async_dispatcher import AsyncDispatcher
+import application_registrar
+import application_summary_registrar
+import deployer_utils
+
+class ApplicationDetailedSummary(object):
+
+    def __init__(self, environment, config):
+        self._environment = environment
+        self._configs = config
+        self._application_registrar = \
+        application_registrar.HbaseApplicationRegistrar(environment['hbase_thrift_server'])
+        self._application_summary_registrar = \
+        application_summary_registrar.HBaseAppplicationSummary(environment['hbase_thrift_server'])
+        self._yarn_connection = \
+        YarnConnection(environment['yarn_resource_manager_host'], \
+            environment['yarn_resource_manager_port'])
+        self._summary_aggregator = ComponentSummaryAggregator()
+        self._component_creators = {}
+        self.dispatcher = AsyncDispatcher()
+
+    def generate(self):
+        """
+        Update applications detailed summary
+        """
+        applist = self._application_registrar.list_applications()
+        logging.info("List of applications: %s", ', '.join(applist))
+        self._application_summary_registrar.sync_with_dm(applist)
+
+        for app in applist:
```

Comment:
Doesn't this mean that every time you call generate, the first few apps will consume the resources in the pool, the rest will be blocked, then next time you call generate the same thing will happen again, starving the remaining applications in the list? I was expecting some kind of circular queue of applications waiting for status updates.

Reply:
To take care of the starvation, we have given 15 seconds of sleep after generating the summary for one round of applications. We will explore the circular queue option to optimize this further.

Reply:
You run through the list in order per call, so if there are 100 applications and 96 don't get resources and it runs again after the first 4 finish, the same first 4 will get the resources again. This simply isn't going to work in normal circumstances - you'll end up with lots of applications never getting updates.